lib/optimize/ConcatenatedModule.js

```diff
@@ -816,6 +810,9 @@ class ConcatenatedModule extends Module {
 			"onsubmit"
 		]);
 
+		// Set of already checked scopes
+		const alreadyCheckedScopes = new Set();
```

Comment:
I think we could use a `WeakSet` here and below since we don't iterate it and it contains objects. `WeakSet` is slightly more performant.